src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,171 @@
+    private final int defaultCapacity = 16;
+    private final float loadFactor = 0.75f;
+    private Node<K, V>[] tables;
+    private int threshold;
+    private int size = 0;
+
+    public MyHashMap() {
+        tables = new Node[defaultCapacity];
+        threshold = (int) (loadFactor * tables.length);
+    }
 
     @Override
     public void put(K key, V value) {
+        Node<K, V> newNode = null;
+        int index = index(key);
+        Objects.checkIndex(index, tables.length);
+        if (key == null) {
+            key = (K) Integer.valueOf(index);
+        }
 
+        if (tables[index] == null) {
+            newNode = new Node<>(index, key, value, null);
+            tables[index(key)] = newNode;
+            if (size++ > threshold) {
+                resize();
+            }
+            return;
+        }
+        newNode = new Node<>(index, key, value, null);
+        if (newNode.getHash() == tables[index].hash) {
+            Node<K, V> tmpTables = tables[index];
+            while (tmpTables != null) {
+                if (newNode.key.equals(tmpTables.key)) {
+                    tmpTables.setValue(newNode.getValue());
+                    return;
+                }
+                tmpTables = tmpTables.next;
+            }
+        }
+        newNode.next = tables[index].next;
+        tables[index].next = newNode;
+        if (size++ > threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        if (key == null) {
+            key = (K) Integer.valueOf(0);
+        }
+        int index = index(key);
+        if (index < 0) {
+            index = index * (-1);
+        }
+        Node<K, V> tmpNode;
+        for (tmpNode = tables[index]; tmpNode != null; tmpNode = tmpNode.next) {
+            if (tmpNode.getKey().equals(key)) {
+                return tmpNode.getValue();
+            }
+
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private Node<K, V>[] resize() {
+        Node<K, V>[] oldTable = tables;
+        Node<K, V>[] newTable = new Node[oldTable.length * 2];
+
+        int index;
+        for (int i = 0; i < oldTable.length; ++i) {
+            Node<K, V> tmp;
+            if ((tmp = oldTable[i]) != null) {
+                oldTable[i] = null;
+                if (tmp.next == null) {
+                    index = tmp.key.hashCode() % newTable.length;
+                    if (index < 0) {
+                        index = index * (-1);
+                    }
+                    newTable[index] = tmp;
+                }
+
+                for (Node<K, V> current = tmp; current != null; current = current.next) {
+                    index = current.key.hashCode() % newTable.length;
+                    if (index < 0) {
+                        index = index * (-1);
+                    }
+                    newTable[index] = tmp;
+
+                }
+            }
+        }
```

Comment:
You can save the old array to a variable, and save new array to the table. And iterate over the old array and call the put method

Reply:
I don't understand what exactly to do if I run through the fact that node.next is not zero and the logic is on those where there is a collision so that they are also thrown ...